[COMSUP-116, COMSUP-152] Node 18 platform support and Astro compatibility (#895)
edgio-docs/edgio-docs: src/guides/v7/performance/cdn_as_code/edgio_config.md

Setting these properties can replicate the behavior of the production environment within your local development workspace.

## cloudRuntime {/* cloudruntime */}

> The `cloudRuntime` key (string) determines which version of Node.js will run your app on our network. Supported values are:

Reviewer: We should indicate that this key requires v7.4.0+.

> The `cloudRuntime` key (string) determines which version of Node.js will run your app on our network. Supported values are:

Author: ```suggestion
The `cloudRuntime` key (string) determines which version of Node.js will run your app on our platform. Supported values are:
```

> The `cloudRuntime` key is a string that denotes the Node.js version for running your app on {{ PRODUCT_NAME }}. Supported values are:

Reviewer: Consider:
```suggestion
The `cloudRuntime` key (string) determines which version of Node.js will run your app on our network. Supported values are:
```

Author: Not sure if "our network" is the right term here since this refers to the Lambda instance for cloud functions.

- `nodejs16.x`
- `nodejs18.x`

> Absent a `cloudRuntime` key, the runtime defaults to the Node.js process version that triggers `{{ CLI_CMD(deploy) }}`. If this version falls outside the supported range, `nodejs18.x` is used. For instance:

Reviewer: Consider:
```suggestion
If the `cloudRuntime` key is not defined, then {{ PRODUCT }} will use the Node.js version from the most recent `{{ CLI_CMD(deploy) }}`. If an unsupported version is detected, then it will set your version to `nodejs18.x`. For instance:
```

Author: I modified this a little from your suggestion. It was confusing to me and made it seem like that it would be based on the last deployment. It is based on the version of Node that executed the deploy command.

> If the `cloudRuntime` key is not defined, then {{ PRODUCT }} will use the Node.js version of the process executing `{{ CLI_CMD(deploy) }}`. If an unsupported version is detected, then it will set your version to `nodejs18.x`. For instance:

Author: ```suggestion
If the `cloudRuntime` key is not defined, then {{ PRODUCT }} will detect your project's Node.js version upon running `{{ CLI_CMD(deploy) }}`. If an unsupported version is detected, then it will set your version to `nodejs18.x`. For instance:
```

```bash
# Here, the cloud runtime is set to nodejs16.x
$ node --version
v16.19.1
$ {{ CLI_CMD(deploy) }}

...

# Here, due to an unsupported version, the cloud runtime defaults to nodejs18.x with a warning
$ node --version
v20.0.0
$ {{ CLI_CMD(deploy) }}
```

<Callout type="important">

> Mismatch between the cloud runtime and the Node.js version used for bundling your project can lead to unexpected behavior. For example, if `cloudRuntime` is set to `nodejs16.x` while the project is bundled with Node.js 18, the project will build with Node.js 18 but run in a Node.js 16 environment. Ensure `cloudRuntime` aligns with the Node.js version used in your project to prevent such issues.

Reviewer: Consider:
```suggestion
  Unexpected behavior may occur when there is a mismatch between your project's Node.js version and the one that runs your app on our network. For example, if the `cloudRuntime` key is set to `nodejs16.x` while the project is bundled with Node.js 18, the project will build with Node.js 18 but run in a Node.js 16 environment. Ensure the `cloudRuntime` key aligns with your project's Node.js version to prevent these types of issues.
```

> Unexpected behavior may occur when there is a mismatch between your project's Node.js version and the one that runs your app on our network. For example, if the `cloudRuntime` key is set to `nodejs16.x` while the project is bundled with Node.js 18, the project will build with Node.js 18 but run in a Node.js 16 environment. Ensure the `cloudRuntime` key aligns with your project's Node.js version to prevent these types of issues.

Author: ```suggestion
Unexpected behavior may occur when there is a mismatch between your project's Node.js version and the one that runs your app on our platform. For example, if the `cloudRuntime` key is set to `nodejs16.x` while the project is bundled with Node.js 18, the project will build with Node.js 18 but run in a Node.js 16 environment. Ensure the `cloudRuntime` key aligns with your project's Node.js version to prevent these types of issues.
```
